pkg/graveler/graveler.go

```diff
@@ -659,6 +665,17 @@ type RefManager interface {
 	// GetBranch returns the Branch metadata object for the given BranchID
 	GetBranch(ctx context.Context, repository *RepositoryRecord, branchID BranchID) (*Branch, error)
 
+	// GetBranchCached gets cached branch information by branch /
+	// repository id.  Cached information is refreshed by time or by
+	// GetBranch.
+	//
+	// Use it only for speedups not for correctness, as it may return
+	// stale (incorrect) data.  The typical flow is to call
+	// GetBranchCached, perform some actions, and commit using an action
+	// that verifies the branch information used was actually correct.
+	// If that fails, retry.
+	GetBranchCached(ctx context.Context, repository *RepositoryRecord, branchID BranchID) (*Branch, error)
```

Comment:
I think we should be more explicit in the documentation about how this is used and how it should not be used. To be on the safe side I would add the documentation to both the declaration and the implementation.